src/helpers/figures/charts/gauge_chart_rendering.ts

```diff
@@ -0,0 +1,466 @@
+import { DEFAULT_FONT } from "../../../constants";
+import { Color, PixelPosition, Rect } from "../../../types";
+import { GaugeChartRuntime } from "../../../types/chart";
+import { relativeLuminance } from "../../color";
+import { clip } from "../../misc";
+import { computeTextWidth, getFontSizeMatchingWidth } from "../../text_helper";
+
+export const GAUGE_PADDING_SIDE = 30;
+export const GAUGE_PADDING_TOP = 10;
+export const GAUGE_PADDING_BOTTOM = 20;
+export const GAUGE_LABELS_FONT_SIZE = 12;
+export const GAUGE_DEFAULT_VALUE_FONT_SIZE = 80;
+
+const GAUGE_BACKGROUND_COLOR = "#F3F2F1";
+export const GAUGE_TEXT_COLOR = "#666666";
+export const GAUGE_TEXT_COLOR_HIGH_CONTRAST = "#C8C8C8";
+const GAUGE_INFLECTION_MARKER_COLOR = "#666666aa";
+const GAUGE_INFLECTION_LABEL_BOTTOM_MARGIN = 6;
+
+export const GAUGE_TITLE_SECTION_HEIGHT = 25;
+export const GAUGE_TITLE_FONT_SIZE = 18;
+export const GAUGE_TITLE_PADDING_LEFT = 10;
+export const GAUGE_TITLE_PADDING_TOP = 5;
+
+interface RenderingParams {
+  width: number;
+  height: number;
+  title: TextProperties;
+  backgroundColor: Color;
+  gauge: {
+    rect: Rect;
+    arcWidth: number;
+    percentage: number;
+    color: Color;
+  };
+  inflectionValues: InflectionValue[];
+  gaugeValue: TextProperties;
+  minLabel: TextProperties;
+  maxLabel: TextProperties;
+}
+
+interface TextProperties {
+  label: string;
+  textPosition: PixelPosition;
+  fontSize: number;
+  color: Color;
+}
+
+interface InflectionValue {
+  rotation: number;
+  label: string;
+  fontSize: number;
+  color: Color;
+  offset: number;
+}
+
+interface UnalignedRectangle {
+  topLeft: PixelPosition;
+  topRight: PixelPosition;
+  bottomRight: PixelPosition;
+  bottomLeft: PixelPosition;
+}
+
+interface Segment {
+  start: PixelPosition;
+  end: PixelPosition;
+}
+
+export function drawGaugeChart(canvas: HTMLCanvasElement, runtime: GaugeChartRuntime) {
+  const canvasBoundingRect = canvas.getBoundingClientRect();
+  canvas.width = canvasBoundingRect.width;
+  canvas.height = canvasBoundingRect.height;
+  const ctx = canvas.getContext("2d")!;
+
+  const config = getGaugeRenderingConfig(canvasBoundingRect, runtime, ctx);
+  drawBackground(ctx, config);
+  drawGauge(ctx, config);
+  drawInflectionValues(ctx, config);
+  drawLabels(ctx, config);
+  drawTitle(ctx, config);
+}
+
+function drawGauge(ctx: CanvasRenderingContext2D, config: RenderingParams) {
+  ctx.save();
+  const gauge = config.gauge;
+
+  const arcCenterX = gauge.rect.x + gauge.rect.width / 2;
+  const arcCenterY = gauge.rect.y + gauge.rect.height;
+  const arcRadius = gauge.rect.height - gauge.arcWidth / 2;
+  if (arcRadius < 0) {
+    return;
+  }
+
+  const gaugeAngle = gauge.percentage === 1 ? 0 : degreesToRadians(180 + 180 * gauge.percentage);
+
+  // Gauge background
+  ctx.strokeStyle = GAUGE_BACKGROUND_COLOR;
+  ctx.beginPath();
+  ctx.lineWidth = gauge.arcWidth;
+  ctx.arc(arcCenterX, arcCenterY, arcRadius, gaugeAngle, degreesToRadians(0));
```

Comment:
degreeToRadians(0) is supposed to be 0 no ?

Reply:
Yeah degreeToRadians(0) is uselsss. For degreeToRadians(180) I left it because for peoples that didn't do math for a long time (not me :roll_eyes: ) 180° is faster to understand than PI